### PR TITLE
Hardcore Accounting

### PR DESCRIPTION
**Pull request by Geevies**
* Ported the accounts terminal UI over to VueUI, replaced the accounts terminal with a modular console.

image
image

One issue it has is that clicking on the account numbers doesn't update the UI. When you close and reopen the UI, it works as expected forever.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/modular_computers/file_system/programs/command/account_database.dm [hunk 1]
```diff
@@ -1,70 +1,81 @@
...
 
-/obj/machinery/account_database/ui_interact(mob/user, ui_key="main", var/datum/nanoui/ui = null, var/force_open = 1)
-	user.set_machine(src)
+	var/obj/item/card/id/held_card = get_held_card()
 
-	var/data[0]
 	data["src"] = "\ref[src]"
+	data["has_printer"] = !!computer.nano_printer
 	data["id_inserted"] = !!held_card
 	data["id_card"] = held_card ? text("[held_card.registered_name], [held_card.assignment]") : "-----"
```
Comment (Karolis2011) [inline code/modules/modular_computers/file_system/programs/command/account_database.dm:80]: Just send over id card field with it's title. Or false if non is present
```suggestion
	data["id_card"] = held_card ? text("[held_card.registered_name], [held_card.assignment]") : FALSE
```

File: code/modules/modular_computers/file_system/programs/command/account_database.dm [hunk 2]
```diff
@@ -91,46 +102,40 @@
...
 			data["transactions"] = trx
 
-	var/list/accounts[0]
+	var/list/accounts = list()
 	for(var/M in SSeconomy.all_money_accounts)
 		var/datum/money_account/D = SSeconomy.get_account(M)
 		accounts.Add(list(list(\
 			"account_number"=D.account_number,\
 			"owner_name"=D.owner_name,\
 			"suspended"=D.suspended ? "SUSPENDED" : "")))
```
Comment (Karolis2011) [inline code/modules/modular_computers/file_system/programs/command/account_database.dm:114]: Just send out all data with transactions and don't bother with `detailed_account_view` and switch between witch account is detailed in UI

File: code/modules/modular_computers/file_system/programs/command/account_database.dm [hunk 3]
```diff
@@ -91,46 +102,40 @@
...
+		return TRUE
 
 	if(href_list["choice"])
 		switch(href_list["choice"])
 			if("create_account")
-				creating_new_account = 1
+				creating_new_account = TRUE
 
 			if("add_funds")
 				var/amount = input("Enter the amount you wish to add", "Silently add funds") as num
```
Comment (Karolis2011) [inline code/modules/modular_computers/file_system/programs/command/account_database.dm:131]: Please remove thoes `input` proc calls and replace them with `<input>` fields in UI itself

File: vueui/src/components/view/mcomputer/command/accountdb.vue [hunk 4]
```diff
@@ -0,0 +1,187 @@
+<template>
+  <div>
+    <div class="item">
+      <div class="itemLabelNarrow">
+        <b>Machine</b>:
+      </div>
+      <div class="itemContent">
+        <span class="average">{{machine_id}}</span>
+      </div>
+    </div>
```
Comment (Karolis2011) [inline vueui/src/components/view/mcomputer/command/accountdb.vue:10]: Please use `<vui-group-item>` and `<vui-group>`

File: vueui/src/components/view/mcomputer/command/accountdb.vue [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    <div class="item">
+      <div class="itemLabelNarrow">
+        <b>ID</b>: {{id_card}}
+      </div>
+    </div>
+
+    <div v-if="access_level > 0">
```
Comment (Karolis2011) [inline vueui/src/components/view/mcomputer/command/accountdb.vue:17]: Don't liter DOM with divs use `<template>` where appropriate.

File: vueui/src/components/view/mcomputer/command/accountdb.vue [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+              <b>Initial Deposit</b>:
+            </div>
+            <div class='itemContent'>
+              <input type='text' id='starting_funds' name='starting_funds'>
+            </div>
+          </div>
+          <div class='item'>
+            <input type='submit' value='Create'>
+          </div>
+        </form>
```
Comment (Karolis2011) [inline vueui/src/components/view/mcomputer/command/accountdb.vue:52]: Use v-model, local variables and `<vui-button>` as final submit

**Review by Karolis2011 — CHANGES_REQUESTED** [review]
Well, this is not great port of said console. DM side is okay for bad port, but VueUI code is awful and bloated.

**Comment by NonQueueingMatt** [comment]
Closed in favour of the other PR.

**Pull request closed without merging**